src/main/java/org/spongepowered/api/text/Texts.java

```diff
@@ -175,25 +203,25 @@ public static Text of() {
      * @return The built text object
      */
     public static Text of(Object... objects) {
+        // Shortcut for lonely TextRepresentables
+        if (objects.length == 1 && objects[0] instanceof TextRepresentable) {
```

Comment:
`TextRepresentable` is basically showing you the same that vanilla does when it tries to display that thing.

> In this case, TextRepresentable hasn't actually helped me achieve my goal of formatting the player, I still have to do it manually every time.

You are mixing things up. The `TextRepresentable` only helps you in just displaying what you have. 

If you want to do anything special then of course you end up doing it yourself.
Oh wait the same applies to your PR as well, but in addition to that you cannot even try to rely on a default behaviour except `toString()`.

If I want to display an ItemStack in the chat I don't want to create a Text for it first. The item stack knows best how to display itself.

>  I have no idea what the TextRepresentable that it extends entails (nothing says so in javadocs),

[TextRepresentable](https://github.com/ST-DDT/SpongeAPI/blob/TextFormatExt/src/main/java/org/spongepowered/api/text/TextRepresentable.java)

> `Gets the textual representation of this instance for its usage in other {@link Text} objects.`

IMO it does, do you have a suggestion to improve that? Or shall I state it in every implementing class what is actualy shown?